portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,56 +14,75 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.data.Post;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import java.util.ArrayList;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Stores and retrieves posts from Datastore */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private ArrayList<String> data = new ArrayList<String>();
-
-  /**
-   * Creates a JSON object where the key, 'messages', is assigned a value of a
-   * variable sized list, containing all strings from the messages parameter.
-   * @param {ArrayList<String>} messages List of messages to be converted.
-   * @return {JSON} Valid JSON object with one key, 'messages', whose value is
-   * the list of strings provided in the messages parameter. 
-   */
-  private static String convertToJson(ArrayList<String> messages) {
-    String json = "{ \"messages\" : [\"";
-    json += String.join("\",\"",messages);
-    json += "\"] }";
-    return json;
-  }
-
   /**
-   * @return {JSON} Valid JSON object with one key, 'messages', whose value is
-   * the list of strings stored in the data ArrayList.
+   * @return {JSON} JSON array with all 'post' entities from Datastore
```

Comment:
You don't need to specify type in the return annotation for Java since it's already a typed language.